netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -79,30 +78,39 @@
    * noticing changes to DNS.
    */
   public static NettyChannelBuilder forAddress(SocketAddress serverAddress) {
-    String authority;
-    if (serverAddress instanceof InetSocketAddress) {
-      InetSocketAddress address = (InetSocketAddress) serverAddress;
-      authority = GrpcUtil.authorityFromHostAndPort(address.getHostString(), address.getPort());
-    } else {
-      // Specialized address types are allowed to support custom Channel types so just assume their
-      // toString() values are valid :authority values. We defer checking validity of authority
-      // until buildTransportFactory() to provide the user an opportunity to override the value.
-      authority = serverAddress.toString();
-    }
-    return new NettyChannelBuilder(serverAddress, authority);
+    return new NettyChannelBuilder(serverAddress);
   }
 
   /**
    * Creates a new builder with the given host and port.
    */
   public static NettyChannelBuilder forAddress(String host, int port) {
-    return new NettyChannelBuilder(
-        new InetSocketAddress(host, port), GrpcUtil.authorityFromHostAndPort(host, port));
+    return forAddress(new InetSocketAddress(host, port));
+  }
+
+  /**
+   * Creates a new builder with the given target URI that will be resolved by
+   * {@link io.grpc.NameResolver}.
+   */
+  public static NettyChannelBuilder forTarget(String targetUri) {
+    return new NettyChannelBuilder(URI.create(targetUri));
```

Comment:
Yes it's going to be handle later, when we add support for the short form target which is by default resolved by DnsNameResolver.